common/messages/heartbeat.go

```diff
@@ -326,7 +327,19 @@ func (m *Heartbeat) FollowerExecute(is interfaces.IState) {
 		if auditServer.GetChainID().IsSameAs(m.IdentityChainID) {
 			if m.IdentityChainID.IsSameAs(is.GetIdentityChainID()) {
 				if m.SecretNumber != is.GetSalt(m.Timestamp) {
-					panic("We have seen a heartbeat using our Identity that isn't ours")
+					lLeaderHeight := is.GetLLeaderHeight()
+					ackChangeValue := is.GetAckChangeValue()
+					ackChangeScheduled := ackChangeValue > is.GetDBHeightAtBoot() && lLeaderHeight <= ackChangeValue
+					if m.DBHeight == lLeaderHeight && (!ackChangeScheduled || is.GetDuplicateIdDetectedAtHeight() > ackChangeValue) {
+						var b strings.Builder
+						fmt.Fprintf(&b, "We have seen a heartbeat using our Identity that isn't ours.\n")
+						fmt.Fprintf(&b, "    Node: %s\n", is.GetFactomNodeName())
+						fmt.Fprintf(&b, "    LLeaderHeight: %d\n", lLeaderHeight)
+						fmt.Fprintf(&b, "    Message dbHeight: %d\n", m.DBHeight)
+						panic(b.String())
```

Comment:
After thinking about it more, we have a feature coming in parchment that allows us to hold a message until a given height. We are working towards a refactor that removes state in areas where it is not needed, so I'd prefer to have this message be held until the height is reached. This removes our height checks here.

Would this also solve the `ackChangeScheduled`? I don't I entirely understand what this logic is aiming for.